### DISCUSSION TITLE
Issues to get started contributing!

### DISCUSSION THREAD

**Discussion by KarinePistili** [ruxailab/RUXAILAB#352] [Announcements]
Hello contributors! We are so happy to having many of you reaching out to us in the past days!! We are creating different issues this month, and accepting many suggestions from you all. 

If you want to get to know the code a little better by developing something, feel free to reach out to any of the issues with the label **help wanted** and **good for wannabe gsocers**.  We will analyse your ideas and assign it to you if it fits.

Right now we are accepting help on RUXAILAB, Figma to Unity VR, Web Eye Tracker and Web Eye Tracker Frontend Feel free to take a look at the issues there ;)

We will publish announcements here, and you are invited to talk about you in the **Introduce Yourself** discussion #307. 

Link to our main repositories:

RUXAILAB (https://github.com/ruxailab/RUXAILAB)
Web Eye Tracker (backend:https://github.com/ruxailab/web-eye-tracker and frontend: https://github.com/ruxailab/web-eye-tracker-front)
Figma VR Unity Converter (https://github.com/ruxailab/figma-vr-unity-converter)

**Some issues guidelines for an easygoing communication**

We work with and assignee system. So you need to express your interest to work on the issue. The first to express interest in the issue comments will have the preference, but be aware to understand what you have to do first and make sure you will have the time and dedication for that before expressing interest, so others also have the chance :wink: 

The person who raised an issue would have the preference to fix it, if the person do not want to or takes too long to answer back, we open it for others.

Let's code! :rocket:

**Comment by sushant4612** [comment]
Hey @KarinePistili , I really want to contribute, but I'm not sure where to start as there are a lot of repositories. I am a newbie and would really appreciate your help.

**Reply by KarinePistili** [reply]
Hello @sushant4612 ! Welcome and thanks for the interest! We have a few projects that we would like contribution now.

They are:
- RUXAILAB (https://github.com/ruxailab/RUXAILAB)
- Web Eye Tracker (backend:https://github.com/ruxailab/web-eye-tracker  and frontend: https://github.com/ruxailab/web-eye-tracker-front)
- Figma VR Unity Converter (https://github.com/ruxailab/figma-vr-unity-converter)

I recommend you to check out for issues with labels such as "Help Wanted", "Good First Issue" or "Good for wannabe gsocers", as we prepared them specially for new contributors who want to star collaborating this year.

I also suggest you to take a look at the ideas list, the proposal template and also to clone the repositories you are interested on.

Feel free to comment on any discussion or issue you want to participate in.

**Reply by sushant4612** [reply]
Thanks for your response, @KarinePistili. 

I have started contributing to RUXAILAB and wanted to set up the `web-eye-tracker` repository. Could you please tell me what I am missing? 

I've sent it in this discussion section: [link to the discussion](https://github.com/orgs/ruxailab/discussions/331#discussioncomment-8763340)

**Reply by KarinePistili** [reply]
Just answered :+1:

**Comment by TheMohit2003** [comment]
Hey @sushant4612, let me share a few tips with you! 😊 Everyone starts as a newbie, so there's no need to stress. 🌟 Fork and clone [this repo](https://github.com/ruxailab/RUXAILAB) and treat it like any other application you use. You'll inevitably encounter various bugs and issues. 🐞 When you do, open an issue detailing your experience, and give fixing it on your forked repo a shot. 🛠️ Once you've resolved the issue, send a PR to the main repo, and voilà! 🎉 You've just become a contributor to this organization. Let's make some awesome contributions together! 🚀

**Reply by sushant4612** [reply]
Thank you for your response and motivation @TheMohit2003 . I have already cloned the repository, and this morning, I raised a pull request and just 2-3 hours ago, that pull request got merged into the repository. 

looking to contribute to more issues

**Comment by ParasPuneetSingh** [comment]
Hi @KarinePistili , I have contributed earlier a bit so I know the basics, I just wanted to know if there is a slack channel or a place where we can interact with the community and ask doubts. I also have experience in ML projects of a small to medium scale. I would like if you could suggest a few starter issues as a start.

**Reply by KarinePistili** [reply]
Hello @ParasPuneetSingh we are creating a Discord server :tada: :tada: . Soon we will share with you all.

**Comment by rohit-sharma-1802** [comment]
Hi @KarinePistili 
Where can I find the issues?

**Reply by KarinePistili** [reply]
Hello @rohit-sharma-1802 , you can find issues that are open to help on the following repositories:

RUXAILAB (https://github.com/ruxailab/RUXAILAB)
Web Eye Tracker (backend:https://github.com/ruxailab/web-eye-tracker and frontend: https://github.com/ruxailab/web-eye-tracker-front)
Figma VR Unity Converter (https://github.com/ruxailab/figma-vr-unity-converter)

**Comment by saltykheera** [comment]
hey @KarinePistili i want to start working on the "Integration of heat maps into Remote Usability LAB".
how do i get started i want to discuss.
![image](https://github.com/ruxailab/RUXAILAB/assets/142778969/c6282a7c-fd74-401d-b494-093c7bcfe6ee)
do i create a repository and start working on it.

**Reply by KarinePistili** [reply]
Hello @saltykheera . First I suggest you to clone https://github.com/ruxailab/RUXAILAB to get to know the project a little better. Then I believe you can research ways of retrieving data such as click and scroll events from the website that is used in the usability test. If you want to do some proof of concept feel free to try it out as well.

**Reply by saltykheera** [reply]
@KarinePistili is there any communication channel were we can ask issues.

**Reply by KarinePistili** [reply]
Do you mean asking for help? We have a discord server and also talk on the discussions here on github.

Link to the discord: https://discord.gg/ErQG6UqR7E

**Comment by DevanshKyada27** [comment]
Hey @KarinePistili I just want to a doubt that is the eye tracking data visualization tool (175h) is been cancelled??  As I want to work on that project.

![image](https://github.com/ruxailab/RUXAILAB/assets/143169520/e51aaf2d-8f08-497a-a4f7-6d2554a07b63)

**Reply by KarinePistili** [reply]
Hello! We prefere to focus the efforts on the other ideas that look more interesting to be done this year. Take a look at the other Eye Tracking idea, it may be interesting for you as well.

**Reply by DevanshKyada27** [reply]
Hey @KarinePistili so can I work on other idea which is based on eye tracking which is not present in the idea list for GSoC 2024. Would it be considerable?

**Reply by KarinePistili** [reply]
Yes, it would be considerable as well. We will go through all proposals sent on gsoc dashboard ;)

**Reply by DevanshKyada27** [reply]
Thank you @KarinePistili :)

**Comment by sitammeur** [comment]
@KarinePistili I am particularly interested in the eye tracking algorithm optimization idea. In discord, I also tagged you to clear up some of my doubts as well. Can you look into that?

**Reply by KarinePistili** [reply]
sure!

**Reply by sitammeur** [reply]
Thanks for the reply I commented regarding the project workflows of the backend in Discord. If I can understand that properly, let's discuss the strategies we can implement. If you want I can mention those in the web eye-tracking discussion as well.

**Reply by KarinePistili** [reply]
it would be great to mention there, so we have all discussion centralized :wink:

**Comment by ConnorMusk** [comment]
Hi everybody! Looking forward t engaging in some discussions with you all!

- Cro

**Comment by xr-dev-saurabh** [comment]
Hi @KarinePistili. I'm Saurabh. Founder and Lead XR Developer at Reealtech, a company specializing in XR product for real estate and hospitality industry. I would like to know more about Figma to Unity VR and Web Eye Tracking. Please suggest good first issues to start with.

**Reply by KarinePistili** [reply]
Hello @xr-dev-saurabh , I'll pin here @JulioManoel for Figma to Unity and @hvini for Eye Tracking project so you can talk a little about the issues open ;)

Usually we would like help on issues with tag **help wanted**, but as we are on gsoc period we also created a **good for wannabe gsocers** which is also good for beginners in the org.

**Comment by samtholathrobin** [comment]
@KarinePistili Ive added a new issue in web-eye-tracker. Could you maybe view the validity of the issue and revert back to me. I would like to start contributing to this repository and help fellow open source contributors.

**Comment by KarinePistili** [comment]
Hello to you all! We added new issues on the projects and will be adding more during the couple weeks. 

**Some issues guidelines for an easygoing communication**

We work with and assignee system. So you need to express your interest to work on the issue. The first to express interest in the issue comments will have the preference, but be aware to understand what you have to do first and make sure you will have the time and dedication for that before expressing interest, so others also have the chance :wink: 

The person who raised an issue would have the preference to fix it, if the person do not want to or takes too long to answer back, we open it for others.

**Comment by monyishh** [comment]
I have set up the codebase in my local system but when I run npm run serve the localhost:8080 page i got is blank white,no error nothing just i can see a blank white page and yellow icon of R on title of site written research-workflow.What should i do?

**Comment by akash-electron** [comment]
>
hey , I  want to  contribute on web Eye tracker frontend

**Comment by vermagaurav24** [comment]
Hello @KarinePistili, I have clone RUXAILAB repo in my local machine but when i try to start server in my local machine it show error. Please guide me how should I setup my project in my local machine so that my server starts and I can start contributing in RUXAILAB repo.

**Reply by karsh0** [reply]
Have you got the fix??

**Reply by TheMohit2003** [reply]
Share the error log here .

**Comment by NarmalaSk** [comment]
Hi @KarinePistili 
’Im Shashi Kumar, a final-year student passionate about open source. I came across this project and found its goal of integrating GitHub contributions with Discord really exciting. However, I couldn’t find a link to the Discord server in the repository, organization, or online resources.

Since the project focuses on automating role management and PR handling via Discord, having access to the server would be essential for understanding the integration better. Could you please provide the Discord server link or clarify how the integration is expected to work?

**Reply by NarmalaSk** [reply]
This project called Integration of github actions with discord role management 2025 gsoc

**Comment by sayandebnath-creator** [comment]
Hi @KarinePistili. I want to contribute to your project, even though it's my first time. I want to give my best to the project you will give me.

**Comment by elenamolero** [comment]
Dear RUXAILAB Team,

My name is Elena Molero, and I am excited to apply for the Google Summer of Code with your organization. I am an spanish final-year Computer Engineering student with a strong artistic vision and an extensive portfolio that combines technology and design.

As an active member of the Aula de Software Libre (Free Software Group) at the University of Córdoba (Spain), I have contributed both technically and artistically to various open-source projects and events, including SalmorejoTECH, one of our most recognized initiatives. My experience spans areas such as front-end development, UI/UX design, and creative programming, making me particularly interested in RUXAILAB’s mission to enhance remote usability testing with AI.

I would love to dedicate 350 hours to a project related to UX/UI, as this is an area I am deeply passionate about. However, I am open to contributing to any project where my skills can be useful. In fact, I have explored past project ideas and found some particularly exciting, such as facial recognition and emotion evaluation, which I would be eager to work on.

I would love to collaborate with your community, learn from your mentors, and contribute meaningfully to the project. I would love to discuss how my skills in Figma, html, css, JavaScript, Flutter, Python, C, C++, Java, React.js... can align with RUXAILAB’s goals.

Looking forward to hearing from you!

Best regards,
Elena :)

**Comment by naohchan** [comment]
Hello! I’m Gerald Rodriguez, and I am currently pursuing an MSc in Computer Science at the University of Bristol in the road to the distinction mark. I have three years of experience in data analytics, reporting, and database management. My technical expertise includes JavaScript, JAVA, Python, C, Git, and software development, with a growing focus on systems programming and automation. In addition I have experienced been Discord Moderator Channel.

I am originally from Peru and now based in Bristol, UK. My motivation to participate in this project came from my passion for open-source software and my interest in contributing to a meaningful initiative, as I have been volunteering in full stack projects in and out of UK.  I see this opportunity as a way to enhance my skills in software development while collaborating with an international community and by doing what I have always like, automate tasks and use Discord!

**Comment by arun-sagar-s-a** [comment]
Hi guys my name is Arun I'm **new to Open Source** and I came across GSoC. I wanted to learn Github Action and I saw the project "**Integration of GitHub Actions with Discord Role Management (90h)**" which gives a motivation to start. Could someone please give me directions on how to start? and where to sign up or anything please. Thank you. Hope to interact with good minds and contribute and learn.

This is my LinkedIn profile: [arun-sagar-s-a](https://www.linkedin.com/in/arun-sagar-s-a-b70b1825b/)

**Comment by LoganSong02** [comment]
Hi @KarinePistili and RUXAILAB team,

I'm Yuhang (Logan) Song, a senior at the University of Southern California majoring in Computer Science. I'm passionate about building impactful solutions that combine software engineering with AI. I've interned at Tencent, worked as a Teaching Assistant for core CS courses at USC, and collaborated with the USC Institute for Creative Technologies to develop an AI-powered math education platform.

I'm proficient in Java, Python, JavaScript, HTML/CSS, React, Spring, MongoDB, MySQL, Redis, Firebase, AWS, and Docker, and I'm certified as an AWS Machine Learning Engineer – Associate. 

I'm especially interested in the following GSoC 2025 projects: UI Layout Optimization for RUXAILAB (175h), Comparative Analysis and Fine-Tuning of Sentiment Models for Improved System Integration (175h), and Integration of GitHub Actions with Discord Role Management (90h).

With my background in frontend&backend development and experience with ML systems, I’m excited to explore how I can contribute meaningfully to your work. Looking forward to learning more and collaborating with the community!

Best,
Logan
Linkedin: https://www.linkedin.com/in/yuhang-song-71415b224/

**Comment by shanilk2004** [comment]
Hi everyone,

I'm Shanil, and I'm new to open source contributions. I always wanted to work on GSoC and got really interested in the project "Automating Issue Creation from SonarQube in CI/CD Pipelines (90h)." It aligns well with what I want to learn, and I believe it would be a great way to start contributing and get to start working in open source contribution.

I’d really appreciate any guidance and the best way to familiarize myself with the codebase. 

Looking forward to learning, collaborating, and contributing! Thank you in advance for your help.

LinkedIn: www.linkedin.com/in/shanil-mohamed-a28ab5281

**Comment by AlwinXue** [comment]
Hi everyone,

I'm Yukai Xue, a master's student in Computer Science at New York University (NYU). I’m passionate about software infrastructure, developer tools, and open-source collaboration. Over the past few years, I’ve completed six software engineering internships at companies including Snowflake, Activision Blizzard, and Philips, focusing on CI/CD automation, monitoring, SonarQube integration, and scalable backend systems.

I’m currently preparing a GSoC proposal for Uramaki LAB, focused on Automating Issue Creation from SonarQube in CI/CD Pipelines. I have previous hands-on experience with SonarQube, GitHub Actions, Jenkins, and Terraform, and I’m excited to apply these skills in an open-source context.

Looking forward to learning from the community and contributing to something impactful!

GitHub: https://github.com/AlwinXue
LinkedIn: https://linkedin.com/in/yukai-xue

**Comment by katerynaloza** [comment]
Hello @KarinePistili ! I am Kateryna Kolobova - a motivated frontend developer - beginner. I am interested in the Uramaki LAB project, and especially UI Layout Optimization for RUXAILAB (175h).

I have knowledge of JavaScript, HTML, CSS, as well as practical skills in working in Figma and creating responsive layouts. In 2022 (with the beginning of the war in Ukraine) I changed the direction of my career and strive to develop in frontend development.

The RUXAILAB project interested me with its goal - to make the interface more accessible, convenient and modern. This is an opportunity for me not only to improve my technical skills, but also to contribute to an important socially oriented initiative.

LinkedIn: https://www.linkedin.com/in/kateryna-kolobova/
GitHub: https://github.com/katerynaloza

**Comment by ChaoCherng-Lee** [comment]
Hello @KarinePistili and everyone else! I am Chao Cherng, a current Software Engineering student heading into my 2nd year. I am interested in contributing to the RUXAILAB project, specifically for the Integration of GitHub Actions with Discord Role Management (90h)!

I'm new to contributing to an open-source project and hope to learn a lot from my mentor. I initially was interested in the following project: "Implementation of A/B Testing Capability in RUXAILAB (175h)", because it aligned more with the course that I was taking, so I figured that I could do and learn more. But since I'm new to this field, I opted for a smaller project, I will still submit my proposal for the Implementation of A/B testing just to test my luck to see if I'd get the chance to contribute!

I know Python, Java, JavaScript, HTML, CSS, C++ & C (mostly self learnt, some learnt through my university units). I recognise this as a chance for me to improve my technical skills as well as getting first hand industry experience by working with a mentor

LinkedIn: https://www.linkedin.com/in/chaochernglee/
Github: https://github.com/ChaoCherng-Lee

**Comment by karsh0** [comment]
Hii everyone, I have setup the repo by following the steps but I'm i am facing some issue with firebase connection can anyone help me with it??

**Comment by 04Soumya** [comment]
Hello RUXAILAB Team,
I have been exploring the GSoC ideas along with the RUXAILAB application and codebase in detail, including the heuristic evaluation workflow, Manager dashboard, Reports, and Final Report screens. Based on this exploration, I would like to propose my approach for enhancing the Playwright testing infrastructure.
From my analysis, the current testing setup has several gaps. There is limited automated coverage for critical user flows such as authentication, study lifecycle, evaluation process, and report generation. The absence of a structured architecture like the Page Object Model makes the test suite harder to maintain and scale. Additionally, there is no automated accessibility validation despite the platform’s focus on usability, cross-browser testing is not consistently configured, and CI/CD integration along with contributor documentation can be significantly improved.
To address these issues, I propose improving the testing system across five key areas. First, I would refactor the existing test suite by introducing the Page Object Model, centralizing fixtures, and replacing unstable selectors with data-testid attributes for better reliability. Second, I would expand test coverage by implementing end-to-end tests for authentication, study lifecycle, evaluator workflows, and report generation including PDF export functionality. Third, I would integrate accessibility testing using @axe-core/playwright to ensure WCAG 2.1 Level AA compliance across major screens. Fourth, I would configure Playwright to run tests across Chromium, Firefox, and WebKit, along with adding mobile viewport testing for responsiveness. Finally, I would improve CI/CD by setting up GitHub Actions workflows to run tests on pull requests and pushes, along with uploading test reports and writing comprehensive contributor documentation.
The implementation would be carried out in structured phases, starting with a codebase audit and setup, followed by architecture refactoring, incremental addition of test coverage, accessibility integration, cross-browser configuration, and finally CI/CD setup and documentation.
The expected outcome of this work would be a maintainable and scalable Playwright test suite using the Page Object Model, high coverage of critical user flows, automated accessibility checks aligned with WCAG standards, reliable cross-browser testing, and a fully integrated CI pipeline. Additionally, clear documentation would help reduce the barrier for new contributors.
I would appreciate your feedback on whether this approach aligns with your expectations for the project. I also had a couple of questions: are there any preferred conventions for introducing data-testid attributes in Vue components, and should accessibility issues identified during testing be tracked as separate GitHub issues or handled within pull requests?
Thank you for your time. I look forward to your feedback and to contributing to RUXAILAB.
Best regards,
Soumya Joshi
[gsoc_proposal_ruxailab_Soumya.pdf](https://github.com/user-attachments/files/26386469/gsoc_proposal_ruxailab_Soumya.pdf)
